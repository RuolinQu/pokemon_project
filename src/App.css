.App {
  text-align: center;
  display: flex;
  justify-content: center;
  width: 100%;
}

.container {
  width: 80%;
  height: 100%;
  display: flex;
  flex-direction: column;
}
.header-container {
  display: flex;
  align-items: center;
  justify-content: left;
}
.searchBar-container {
  display: flex;
  align-items: center;
  justify-content: right;
}
img {
  height: 300px;
  width: 250px;
}
.searchButton {
  background-color: rgb(83, 125, 68);
  color: white;
  border-radius: 0px;
}
.content-section {
  margin: 20px 0;
  display: flex;
  flex-direction: row;
  gap: 20px;
  justify-content: space-between;
}

.card-header {
  padding: 0 20px;
  background-color: #2b5543;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.card-container:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}
.card-container {
  border-radius: 2px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  width: 40%;
}
.pageButton {
  border: 1px solid;
  background-color: white;
  border-color: black;
}

.footer {
  display: flex;
  justify-content: space-between;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
