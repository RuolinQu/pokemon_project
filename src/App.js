import logo from "./logo.svg";
import Header from "./components/Header/Header";
import "./App.css";
import SearchBar from "./components/SearchBar/SearchBar";
import { fetchAllData, fetchData } from "./api/api";
import { useEffect, useState } from "react";
import PokemonCard from "./components/PokeCard/PokeCard";

function App() {
  const [searchInput, setSearchInput] = useState("");
  const [data, setData] = useState([]);
  const [pageNumber, setPageNumber] = useState(0);
  const [pageCount, setPageCount] = useState(2);
  useEffect(() => {
    const helper = async () => {
      const allData = await fetchAllData(pageNumber);
      setPageCount(Math.ceil(allData.count / 3));
      Promise.all(
        allData.results.map(async (item) => {
          const response = await fetch(item.url);
          const itemData = await response.json();
          return {
            name: item.name,
            id: itemData.id,
            src: itemData.sprites.other["dream_world"]["front_default"],
          };
        })
      ).then((data) => {
        setData(data);
      });
    };
    helper();
  }, [pageNumber]);

  useEffect(() => {
    const helper = async () => {
      const allData = await fetchAllData(pageNumber);
      setPageCount(Math.ceil(allData.count / 3));
      Promise.all(
        allData.results.map(async (item) => {
          const response = await fetch(item.url);
          const itemData = await response.json();
          return {
            name: item.name,
            id: itemData.id,
            src: itemData.sprites.other["dream_world"]["front_default"],
          };
        })
      ).then((data) => {
        setData(data);
      });
    };
    helper();
  }, []);

  const onChangePage = (isNext) => {
    setPageNumber(prev=>isNext?(prev+1):(prev-1))
  }

  const handleInput = (e) => {
    setSearchInput(e.target.value);
  };
  const handleClick = async () => {
    if (searchInput) {
      const data = await fetchData(searchInput);
      setPageCount(0);
      setData([
        {
          name: data.species.name,
          id: data.id,
          src: data.sprites.other["dream_world"]["front_default"],
        },
      ]);
    } else {
      const allData = await fetchAllData(pageNumber);
      console.log("alldata", allData);
      setPageCount(Math.ceil(allData.count / 3));
      Promise.all(
        allData.results.map(async (item) => {
          const response = await fetch(item.url);
          const itemData = await response.json();
          return {
            name: item.name,
            id: itemData.id,
            src: itemData.sprites.other["dream_world"]["front_default"],
          };
        })
      ).then((data) => {
        setData(data);
      });
    }
  };
  return (
    <div className="App">
      <div className="container">
        <Header />
        <SearchBar
          searchInput={searchInput}
          handleInput={handleInput}
          handleClick={handleClick}
        />
        <section className="content-section">
          {data.map((item) => (
            <PokemonCard data={item} />
          ))}
        </section>
        <footer className="footer">
          <button disabled={pageNumber <= 0} className="pageButton" onClick={()=>onChangePage(0)}>
            &#8592;previous page
          </button>
          <button disabled={pageNumber >= pageCount} className="pageButton" onClick={()=>onChangePage(1)}>
            next page&#8594;
          </button>
        </footer>
      </div>
    </div>
  );
}

export default App;
