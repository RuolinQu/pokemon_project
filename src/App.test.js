import { render, screen } from '@testing-library/react';
import { Item } from 'semantic-ui-react';
import App from './App';



describe("app", () => {
  it("should render snapshot", () => {
    const wrapper = render(<App />);
    const firstRender = wrapper.asFragment();
    expect(firstRender).toMatchSnapshot();
  })
  it("should render header", () => {
    const wrapper = render(<App />);
    const headerElement = wrapper.queryByText("Pokemon Search");
    expect(headerElement).toBeInTheDocument()
  })
})
