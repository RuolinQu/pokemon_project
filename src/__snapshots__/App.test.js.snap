// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <div
        class="header-container"
        id="header"
      >
        <h2
          style="color: rgb(69, 68, 69);"
        >
          Pokemon Search
        </h2>
      </div>
      <div
        class="searchBar-container"
      >
        <input
          value=""
        />
        <button
          class="searchButton"
        >
          Search
        </button>
      </div>
      <section
        class="content-section"
      />
      <footer
        class="footer"
      >
        <button
          class="pageButton"
          disabled=""
        >
          ←previous page
        </button>
        <button
          class="pageButton"
        >
          next page→
        </button>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
